internal/provider/data_source_http_test.go

```diff
@@ -471,6 +472,36 @@ func TestDataSource_UnsupportedInsecureCaCert(t *testing.T) {
 	})
 }
 
+func TestDataSource_HTTPViaProxyWithEnv(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer svr.Close()
+
+	p := goproxy.NewProxyHttpServer()
+
+	proxy := httptest.NewServer(p)
+	defer proxy.Close()
```

Comment:
Over in the terraform-provider-tls acceptance testing, all the testing HTTP servers, including the proxies, have connection counting:

https://github.com/hashicorp/terraform-provider-tls/blob/b9cbb62bc2a8b90f6af91e86047082a5357e8c2f/internal/provider/testutils/local_server.go#L38-L46

Then there is a `TestCheckFunc` which ensures that the "real" HTTP server and proxy HTTP server have matching connection counts:

https://github.com/hashicorp/terraform-provider-tls/blob/b9cbb62bc2a8b90f6af91e86047082a5357e8c2f/internal/provider/testutils/local_server.go#L117-L127

I think we'll need something similar here to verify that the proxy is correctly being used, unless there is a better testing strategy.

Reply:
I'm looking into the current test failures (e.g., https://github.com/hashicorp/terraform-provider-http/actions/runs/3376234811/jobs/5603729486).
```
Check 2/2 error: expected server and proxy active connection count to match: server was 1, while proxy was 0
```

I can't reproduce these locally, I see a different failure.
```
Check 2/2 error: expected server and proxy active connection count to match: server was 3, while proxy was 1
```

Reply:
I see the same failures in CI (e.g., https://github.com/hashicorp/terraform-provider-http/actions/runs/3376368850/jobs/5604018894) when using 
```
	tr := &http.Transport{
		Proxy: http.ProxyFromEnvironment,
	}
```

But all tests pass when running locally with this configuration.